CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,75 @@
+# Code of Conduct
+
+## When something happens
+
+If you see a violation of our community's standards of [Conduct](#conduct), follow these steps:
+
+1. Let the person know that what they did is not appropriate and ask them to stop and/or edit their messages.
+2. That person should immediately stop the behavior and correct the issue.
+3. If this doesn’t happen, or if you're uncomfortable speaking up, contact the maintainers.
+4. As soon as available, a maintainer will look into the issue, and take further action, starting with a warning, then temporary block, then long-term repo or organization ban.
+
+When reporting, please include any relevant details, links, screenshots, context, or other information that may be used to better understand and resolve the situation.
+
+The maintainers will prioritize the well-being and comfort of the recipients of the violation over the comfort of the violator.
+
+## Conduct
+
+_This section is based on the [Rust Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct)._
+
+* We are committed to providing a friendly, safe and harassment-free environment for all, regardless of level of experience, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion, nationality, or other similar characteristic.
+* Please avoid using overtly sexual aliases or other nicknames that might detract from a friendly, safe and welcoming environment for all.
+* Please be kind and courteous. There’s no need to be mean or rude.
+* Respect that people have differences of opinion and that every design or implementation choice carries a trade-off and numerous costs. There is seldom a right answer.
+* Please keep unstructured critique to a minimum. If you have solid ideas you want to experiment with, make a fork and see how it works.
+* We will exclude you from interaction if you insult, demean or harass anyone. That is not welcome behavior. Please read our definition of [Harassment](#harassment). In particular, we don’t tolerate behavior that excludes people in socially marginalized groups.
+* Private harassment is also unacceptable. No matter who you are, if you feel you have been or are being harassed or made uncomfortable by a community member, please contact the maintainers immediately. Whether you’re a regular contributor or a newcomer, we care about making this community a safe place for you and we’ve got your back.
+* Please do not try to present "reverse-ism" versions of the conduct violations that apply specifically to underrepresented groups. Examples of reverse-isms are "reverse racism", "reverse sexism", "heterophobia", and "cisphobia".
```

Comment:
This line might be a cause of contention.

`@lon` in Discord raised a valid point:

> I have a serious concern about this line. It seems needlessly divisive to me. Harassment is harassment; it is not a function of the target's identity.
> In my eyes, it's easy to interpret this line as justifying certain forms of harassment.

> this suggestion still perpetuates division by creating rules that only apply to certain groups. This line suggests to me that some members of the community do not deserve the same protection as others. Ideally, I would like to see everyone held to the same standards and granted the same protections."

My view is that it is an important point not to allow a kind of false equivalency that can be (and has been in other communities) weaponised against underrepresented groups, so that is what this is trying to address.

Reply:
There is a good article on "reverse racism" here: https://en.wikipedia.org/wiki/Reverse_racism

Reply:
I highly disagree with this line. "Reverse racism" is just another term for racial discrimination, and there is certainly plenty of racial discrimination in the world. Someone very close to me has been discriminated against, threatened, and excluded from a lot of things for being white, and any community which encourages this kind of behavior is a community that I would not contribute to or support in any way.

Political opinions in a CoC will only lead to division and will discourage people (like myself) from joining and contributing to the community. I think this line should be removed.

Reply:
The point is not to deny that white people too can face discrimination. Please read the wikipedia page a bit more carefully: it's about people using racism against white people to argue against, among others, groups working for equal treatment of minorities.

If someone comes along and says: "This whole trans people's rights is bullshit, cisgender people too face discrimination"
well that's obviously not ok

Reply:
Thanks @TehPers.

> Political opinions in a CoC will only lead to division

I'm not sure what an _apolitical_ CoC and moderation policy would look like and why that would be desirable. IMHO any code of conduct is inherently political, at least from a certain point of view, and that is a good and generally important thing for any community to work and for the safety of its members.

Reply:
I think it is important for the maintainers/moderators of this community to take a political stance, especially given the "Keep politics out of my video games!" viewpoint that is commonly held within the gaming community – this can be really toxic and needs to be dealt with upfront.

Reply:
> I'm not sure what an apolitical CoC and moderation policy would look like and why that would be desirable. IMHO any code of conduct is inherently political, at least from a certain point of view, and that is a good and generally important thing for any community to work and for the safety of its members.

Certain parts of a code of conduct may seem political (such as being inclusive of LGBTQ+ members), but are generally included for non-political reasons, such as wanting to be inclusive of all kinds of people. On the other hand, points which are strictly added for political reasons cause anybody who disagrees with those political points to be excluded from the community for no reason other than that they have a different political point of view. Should a person be excluded because they are a Libertarian, Republican, or because they supported Brexit? I don't think so, and I think that including points like this just leads to more discussion about politics on a project that has nothing to do with politics. "Inclusion" means being inclusive of all kinds of people, including the people you disagree with.

Reply:
> On the other hand, points which are strictly added for political reasons cause anybody who disagrees with those political points to be excluded from the community for no reason other than that they have a different political point of view

I don't think anything in the current draft is anything like that

Reply:
Thanks for explaining @TehPers – it is at the discretion of the maintainers to decide which viewpoints are considered acceptable and which are not. Homophobia, for example, is a political viewpoint that is probably not considered to be acceptable, whereas wanting to leave the EU is acceptable.

Reply:
FWIW, my intention was not to add things here for political reasons, but purely because I believe similar statements have been important in other communities.

Reply:
> Homophobia, for example, is a political viewpoint that is probably not considered to be acceptable

Homophobia is also exclusive of a group of people. It's a politicized topic, but should not be considered acceptable because it's non-inclusive.

> I understand your point, and also don't think anything in the current draft is anything like that, do you agree with this statement?

No. If I agreed with this statement, I wouldn't have any issue with this particular line in the draft.

Reply:
Apologies if I was unclear before – I think allowing claims of "reverse racism" to pass as valid complaints is non-inclusive in _exactly the same way_ as homophobia. I would like to do a better job of explaining why this is the case, but I'm unfortunately not an expert on racism, so I can only suggest reading what others have written on the subject.

Reply:
I can provide an example which is closer to home for me. If a community were to allow the complaint "TERF is a slur" to be taken as a valid harassment claim, I would not feel able to participate in that community because I am trans. This is the point that the "reverse-isms" clause aims to address.

Reply:
I've removed this line based on discussion in the Discord channel. Thanks @TehPers.